cli/flox-rust-sdk/src/flox.rs

```diff
@@ -206,6 +206,50 @@ impl Flox {
 pub static DEFAULT_FLOXHUB_URL: Lazy<Url> =
     Lazy::new(|| Url::parse("https://hub.flox.dev").unwrap());
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct FloxhubToken(String);
+
+impl AsRef<str> for FloxhubToken {
+    fn as_ref(&self) -> &str {
+        &self.0
+    }
+}
+
+impl FloxhubToken {
+    /// Create a new floxhub token from a string
+    pub fn new(token: String) -> Self {
+        FloxhubToken(token)
+    }
+
+    /// Return the token as a string
+    pub fn secret(&self) -> &str {
+        &self.0
+    }
+
+    pub fn handle(&self) -> Result<String, FloxhubTokenError> {
+        #[derive(Debug, Deserialize)]
+        struct Claims {
+            #[serde(rename = "https://flox.dev/handle")]
+            handle: String,
+        }
+
+        let mut validation = Validation::default();
+        // we're neither creating or verifying the token on the client side
```

Comment:
nonblocking: why not?

Reply:
the token is signed and validated on the server side using a secret key (`"AAAA.."` in the fake token above.
if the claims are forged its going to be the server that validates the token and trips over the signature.